Makefile

```diff
@@ -9,6 +9,27 @@ gen-mock:
 	@go install github.com/gojuno/minimock/v3/cmd/minimock@v3.3.13
 	@go generate -run minimock ./...
 
+# For the future compogen developer, they can use this command to generate the documentation for the component by modified compogen tool.
+# t stands for type of component and c stands for component name
+# Example: make local-gen-doc t=application c=slack
+local-gen-doc:
```

Comment:
I took out `rm`.

Thanks for your advise!
Now, I feel compogen is customised for component repo. So, it means they are not independent essentially.
In this case, I do not think we will extract compogen.
So, remaining it in here may make senses now.
How about we remove it if we decide to move compogen out?

Reply:
I agree with you `compogen` (component generation, for now docs but later perhaps it is used for boilerplate generation) is coupled with `component`. What I meant is that this coupling shouldn't be bi-direcional. So, you're right, it might not happen that we extract it because there won't be other packages using it. But we should keep it self-contained.

I'm thinking that there's coupling in the other direction, too, in the tags that we have in each `main.go` for generating the docs. So I guess my thought goes: I'd like to keep the component <- compogen coupling at the minimum because one day we might want to use a different tool or strategy (AI generating the docs from the source code? 😏 just fantasising) and I'd like to make that potential transition painless.

Reply:
>  I'd like to keep the component <- compogen coupling at the minimum

Good point!

> the alternative I'm suggesting (running this make command from the compogen package) is a simple change and it might make more sense if it's intended for the dev modifying compogen. From tools/compogen/Makefile

I will follow this suggestion! Thanks!